src/Build.UnitTests/Construction/SolutionFile_Tests.cs

```diff
@@ -2400,5 +2400,42 @@ public void ParseSolutionFileContainingProjectsWithSimilarNames_FourProjects_One
 
             exception.Message.ShouldStartWith(message);
         }
+
+        /// <summary>
+        /// A test where paths contain ..\ segments to ensure the paths are normalized.
+        /// </summary>
+        [Fact]
+        public void ParseSolutionWithParentedPaths()
+        {
+            string solutionFileContents =
+                @"
+                Microsoft Visual Studio Solution File, Format Version 9.00
+                # Visual Studio 2005
+                Project('{749ABBD6-B803-4DA5-8209-498127164114}')  = 'ProjectA',  '..\ProjectA\ProjectA.csproj', '{0ABED153-9451-483C-8140-9E8D7306B216}'
+                EndProject
+                Global
+                    GlobalSection(SolutionConfigurationPlatforms) = preSolution
+                        Debug|AnyCPU = Debug|AnyCPU
+                        Release|AnyCPU = Release|AnyCPU
+                    EndGlobalSection
+                    GlobalSection(ProjectConfigurationPlatforms) = postSolution
+                        {0ABED153-9451-483C-8140-9E8D7306B216}.Debug|AnyCPU.ActiveCfg = Debug|AnyCPU
+                        {0ABED153-9451-483C-8140-9E8D7306B216}.Debug|AnyCPU.Build.0 = Debug|AnyCPU
+                        {0ABED153-9451-483C-8140-9E8D7306B216}.Release|AnyCPU.ActiveCfg = Release|AnyCPU
+                        {0ABED153-9451-483C-8140-9E8D7306B216}.Release|AnyCPU.Build.0 = Release|AnyCPU
+                    EndGlobalSection
+                    GlobalSection(SolutionProperties) = preSolution
+                        HideSolutionNode = FALSE
+                    EndGlobalSection
+                EndGlobal
+                ";
+
+            SolutionFile solution = ParseSolutionHelper(solutionFileContents);
+            string expectedRelativePath = NativeMethodsShared.IsWindows ? @"..\ProjectA\ProjectA.csproj" : @"../ProjectA/ProjectA.csproj";
+            Assert.Equal("ProjectA", solution.ProjectsInOrder[0].ProjectName);
```

Comment:
nit: Shouldly?
```suggestion
            solution.ProjectsInOrder[0].ProjectName.ShouldBe("ProjectA");
```
etc.

Reply:
Should I convert this whole file to Shouldly so its consistent?

Reply:
I wouldn't push for it. My thought is: Shouldly on new tests and when modifying tests --> it will eventually be used in all tests in the MSBuild repo that have the possibility of failing. If you want to convert the whole file, though, I certainly wouldn't stop you. (I did that at one point when I was frustrated and wanted something easy and monotonous.)

Reply:
I'll take a stab at converting the whole file.  I personally like consistency so I'd rather have all older Asserts or Shouldly, no mix...

Reply:
Done, soooooo much better!